### PR TITLE
orfp64a32: Add new GCC9 bit offset instruction encoding

### PR DESCRIPTION
**Pull request by stffrdhrn**
As suggested by RTH, we rather use some bits to indicate the 2 ro 1
index offset from register 1.  Using the reserved bits as per proposal:

  - https://openrisc.io/proposals/orfpx64a32

Tested this with a simple fp c program (thats how I found I needed the reserved bits to not be validated).

```
fusesoc run --target marocchino_tb mor1kx-generic --elf-load ./hellofp --option_orfpx64a32_abi GCC9
....

elf-loader: /home/shorne/work/gnu-toolchain/hellofp was loaded
Loading       21870 words
                   0 : Illegal Wishbone B3 cycle type (xxx)
hello gcc from or1k fpu: 1.200000 * 2 = 2.400000: ok.
test: gt 1
test: ge 1
test: ge 1
test: lt 1
test: le 1
test: le 1
test: ne 1
test: eq 1
Branching: ok.

hello gcc from or1k fpu: 134.322006 * 2 = 268.644012: ok.
testd: gt 1
testd: ge 1
testd: ge 1
testd: lt 1
testd: le 1
testd: le 1
testd: ne 1
testd: eq 1
D-branching: ok.

```

### CODE REVIEW

**Comment by bandvig** [comment]
It looks good. 

However I would propose to create PR for branch `fp_unordered_cmp` in `openrisc/or1k_marocchino` (https://github.com/openrisc/or1k_marocchino/tree/fp_unordered_cmp). Even so CI will fail, I think the branch could be good place for gathering all FP-related updates. As CI tools be updated to spec 1.3 we just merge whole bundle to `master`.

Have you  tried to build NewLIB with you spec-1.3 GCC9? Yesterday I build yours `binutils-gdb/tree/orfpx64a32-3` and `gcc/tree/or1k-fpu-2` (stage-1). Than I tried to build NewLIB with `-mhard-float -mdouble-float` options and got error message from assembler. I could not reproduce the message exactly right now as I'm out of home, but I remember that one of register pair looked like `r31,r16` in double instruction caused the error.

**Comment by stffrdhrn** [comment]
I thought I tried to build newlib as well. Let me try again. I'll move the
pr to your unordered Branch.  You can move it too if you get there before
me. I'll be out of town tomorrow.

**Comment by bandvig** [comment]
@stffrdhrn 
GitHub’s web-interface allows change branch of my repo for merging but desn’t allow to change repo. I just copy-pasted your changes into `or1k_marocchino/tree/fp_unordered_cmp`.
